loader/standard_loader.py

```diff
@@ -13,58 +11,125 @@
+                                '(?P<secfrac>\.[0-9]+)?'
+                                '(Z|(\+|-)(?P<offset_hour>[01][0-9]|2[0-3]):(?P<offset_minute>[0-5][0-9]))?$')
 
     def __init__(self, dss_uploader: DssUploader, metadata_file_uploader: MetadataFileUploader) -> None:
         self.dss_uploader = dss_uploader
         self.metadata_file_uploader = metadata_file_uploader
-
-    @staticmethod
-    def _parse_bundle(bundle: dict) -> ParsedBundle:
-        try:
-            data_bundle = bundle['data_bundle']
-            bundle_uuid = data_bundle['id']
-            metadata_dict = data_bundle['user_metadata']
-            data_objects = bundle['data_objects']
-        except KeyError:
-            # TODO: Make a bundle parse error that spits back whatever info we can get about the bundle
-            raise
-        return ParsedBundle(bundle_uuid, metadata_dict, data_objects)
+        # these will probably need to be made into queues for parallelization
+        self.bundles_parsed: typing.List[ParsedBundle] = []
+        self.bundles_failed_unparsed: typing.List[dict] = []
+        self.bundles_loaded: typing.List[ParsedBundle] = []
+        self.bundles_failed_parsed: typing.List[ParsedBundle] = []
 
     @classmethod
     def _get_file_uuid(cls, file_guid: str):
         result = cls._uuid_regex.findall(file_guid.lower())
         if result is None:
-            raise ValueError(f'Misformatted file_guid: {file_guid} should contain a uuid.')
+            raise ParseError(f'Misformatted file_guid: {file_guid} should contain a uuid.')
         if len(result) != 1:
-            raise ValueError(f'Misformatted file_guid: {file_guid} contains multiple uuids. Only one was expected.')
+            raise ParseError(f'Misformatted file_guid: {file_guid} contains multiple uuids. Only one was expected.')
         return result[0]
 
-    @staticmethod
-    def _get_file_version(file_info: dict):
+    @classmethod
+    def _get_file_version(cls, file_info: dict):
         """Since date updated is optional, we default to date created when it's not updated"""
-        try:
-            return file_info['updated']
-        except KeyError:
-            return file_info['created']
+        def parse_version_key(file_info_, key):
+            """return None if version cannot be found"""
+            try:
+                match = cls._rfc3339_regex.fullmatch(file_info_[key])
+                if match is None:
+                    logger.warning(f'Failed to parse file version from date {key}: {file_info_[key]}')
+                    return None
+                return file_info_[key]
+            except KeyError:
+                return None
+        version = parse_version_key(file_info, 'updated')
+        if version is None:
+            version = parse_version_key(file_info, 'created')
+        if version is None:
+            raise ParseError('Either bundle had no updated / created time or it was not rfc3339 compliant')
```

Comment:
It seems like knowing which of these cases occurred (not present or invalid format) would be useful.
However, for current purposes, I think addressing this is optional.

Reply:
a warning is logged which I think makes this clear in practice.